File: Sources/BraintreeThreeDSecure/V2UICustomization/BTThreeDSecureV2ButtonCustomization.m
#import "BTThreeDSecureV2BaseCustomization_Internal.h"

#if __has_include(<Braintree/BraintreeThreeDSecure.h>)
#import <Braintree/BTThreeDSecureV2ButtonCustomization.h>
#else
#import <BraintreeThreeDSecure/BTThreeDSecureV2ButtonCustomization.h>
#endif

@implementation BTThreeDSecureV2ButtonCustomization

- (instancetype)init {
    self = [super init];
    if (self) {
        self.cardinalValue = [ButtonCustomization new];
    }

    return self;
}

- (void)setBackgroundColor:(NSString *)backgroundColor {
    _backgroundColor = backgroundColor;
    ((ButtonCustomization *)self.cardinalValue).backgroundColor = backgroundColor;
}

- (void)setCornerRadius:(int)cornerRadius {
    _cornerRadius = cornerRadius;
    ((ButtonCustomization *)self.cardinalValue).cornerRadius = cornerRadius;
}

@end

File: Sources/BraintreeThreeDSecure/V2UICustomization/BTThreeDSecureV2ButtonCustomization.swift
import Foundation
import CardinalMobile

/// Button customization options for 3D Secure 2 flows.
@objcMembers public class BTThreeDSecureV2ButtonCustomization: BTThreeDSecureV2BaseCustomization {

    // MARK: - Public Properties

    /// Color code in Hex format. For example, the color code can be “#999999”.
    public var backgroundColor: String? {
        get { _backgroundColor }
        @objc(setBackgroundColor:) set {
            _backgroundColor = newValue
            (cardinalValue as? ButtonCustomization)?.backgroundColor = newValue
        }
    }

    /// Radius (integer value) for the button corners.
    public var cornerRadius: Int {
        get { _cornerRadius }
        @objc(setCornerRadius:) set {
            _cornerRadius = newValue
            (cardinalValue as? ButtonCustomization)?.cornerRadius = Int32(newValue)
        }
    }

    // MARK: - Private Properties

    /// Used as a holder for Obj-C interoperability
    private var _backgroundColor: String?

    /// Used as a holder for Obj-C interoperability
    private var _cornerRadius: Int = 0

    // MARK: - Initializer

    public override init() {
        super.init()
        cardinalValue = ButtonCustomization()
    }
}
